hw/dv/dpi/usbdpi/usbdpi.c

```diff
@@ -92,10 +93,10 @@ void *usbdpi_create(const char *name, int loglevel) {
 
 const char *decode_usb[] = {"SE0", "0-K", "1-J", "SE1"};
 
-void usbdpi_device_to_host(void *ctx_void, svBitVecVal *d2p_data) {
+void usbdpi_device_to_host(void *ctx_void, const svLogicVecVal *usb_d2p) {
```

Comment:
Is the change of data type from `svBitVecVal` to `svLogicVecVal` really needed here, is there any particular benefit? I am not against it but it does not seem necessary as the code below only uses the `aval` struct member of `usb_d2p`. The `bval` member indicating if a bit is `Z` or `X` is not ever looked at. In fact, this may lead to other issues. 

Alternatively, the argument in `usbdpi.sv` should be changed from `input [4:0] d2p` to `input bit[4:0] d2p` instead. Or would this lead to the same problem in case argument contains `X` and `Z`?

Maybe @imphil could provide some input here.

Reply:
As suggested by @vogelpi, I'd go with bit instead of logic vectors here; that's also consistent with our definition of `p2d`.

- Change `input [4:0] d2p` to `input bit [4:0] d2p`. That should already be enough most likely.
- If not, modify `logic [4:0] d2p;` to `bit [4:0] d2p;` as well.

Reply:
Thinking back, I believe I couldn't find good documentation on what types were preferred so there was some copy/paste programming used here.

Reply:
Ok, I've changed the function parameters (back) to svBitVecVal and instead changed the parameter in usbdpi.sv to bit.

Reply:
Thanks. Also this is resolved.